packages/eslint-plugin-next/lib/rules/no-script-bi-outside-document.js

```diff
@@ -0,0 +1,51 @@
+const path = require('path')
```

Comment:
In addition to an ESLint warning, would be good to have a runtime console warning since some people don't use ESLint and this will actually break once streaming is added.

Reply:
I am not sure if theres an easy way to do this. We will have to know the component's parent. I tried using `React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner` to get that info, but that didn't work. Any advice is appreciated @kara @devknoll

Reply:
@janicklas-ralph `Document` and `App` are rendered in separate React trees, so I think you could just reuse the context to determine if we're rendering `Document` or not and warn appropriately

Reply:
+1. I think it's OK to put the console warning in a separate PR though (I'm good with merging this one with just the ESLint warning and docs)